src/Symfony/Component/Yaml/Inline.php

```diff
@@ -604,10 +604,19 @@ private static function evaluateScalar($scalar, $flags, $references = array())
                         return self::evaluateBinaryScalar(substr($scalar, 9));
                     case preg_match('/^(-|\+)?[0-9,]+(\.[0-9]+)?$/', $scalar):
                         return (float) str_replace(',', '', $scalar);
-                    case preg_match(self::getTimestampRegex(), $scalar):
+                    case preg_match(self::getTimestampRegex(), $scalar, $datetimeMatches):
                         if (Yaml::PARSE_DATETIME & $flags) {
-                            $date = new \DateTime($scalar);
-                            $date->setTimeZone(new \DateTimeZone('UTC'));
+
+                            // Timestamps that have timezone data should be parsed in the specified timezone first, then
+                            // converted to UTC; if no tz data is supplied, parse as UTC date initially
+                            if (isset($datetimeMatches['tz'])) {
+                                $date = new \DateTime($scalar);
```

Comment:
The timezone data is in the string (`$scalar`) so no need to add it to the constructor.